packages/embark-ipfs/src/storageProcessesLauncher.js

```diff
@@ -0,0 +1,187 @@
+import { __ } from 'embark-i18n';
+import { joinPath, canonicalHost, buildUrlFromConfig } from 'embark-utils';
+import { ProcessLauncher } from 'embark-core';
+const shellJs = require('shelljs');
+const constants = require('embark-core/constants');
+const cloneDeep = require('lodash.clonedeep');
+
+let References = {
+  ipfs: 'https://ipfs.io/docs/install/',
+  swarm: 'http://swarm-guide.readthedocs.io/en/latest/installation.html'
+};
+
+class StorageProcessesLauncher {
+  constructor(options) {
+    this.logger = options.logger;
+    this.events = options.events;
+    this.storageConfig = options.storageConfig;
+    this.webServerConfig = options.webServerConfig;
+    this.blockchainConfig = options.blockchainConfig;
+    this.embark = options.embark;
+    this.processes = {};
+    this.corsParts = options.corsParts || [];
+    this.restartCalled = false;
+    this.manualExit = false;
+
+    this.cors = this.buildCors();
+
+    this.events.on('exit', () => {
+      Object.keys(this.processes).forEach(processName => {
+        this.processes[processName].send('exit');
+      });
+    });
+  }
+
+  buildCors() {
+    let corsParts = cloneDeep(this.corsParts);
+    // add our webserver CORS
+    if (this.webServerConfig.enabled) {
+      if (this.webServerConfig && this.webServerConfig.host) {
+        corsParts.push(buildUrlFromConfig(this.webServerConfig));
+      }
+      else corsParts.push('http://localhost:8000');
+    }
+
+    // add all dapp connection storage
+    if (this.storageConfig.enabled) {
+      this.storageConfig.dappConnection.forEach(dappConn => {
+        if (dappConn.getUrl || dappConn.host) {
+
+          // if getUrl is specified in the config, that needs to be included in cors
+          // instead of the concatenated protocol://host:port
+          if (dappConn.getUrl) {
+            // remove /ipfs or /bzz: from getUrl if it's there
+            let getUrlParts = dappConn.getUrl.split('/');
+            getUrlParts = getUrlParts.slice(0, 3);
+            let host = canonicalHost(getUrlParts[2].split(':')[0]);
+            let port = getUrlParts[2].split(':')[1];
+            getUrlParts[2] = port ? [host, port].join(':') : host;
+            corsParts.push(getUrlParts.join('/'));
+          }
+          // in case getUrl wasn't specified, use a built url
+          else {
+            corsParts.push(buildUrlFromConfig(dappConn));
+          }
+        }
+      });
+    }
+
+    if (this.blockchainConfig.enabled) {
+      // add our rpc endpoints to CORS
+      if (this.blockchainConfig.rpcHost && this.blockchainConfig.rpcPort) {
+        corsParts.push(`http://${canonicalHost(this.blockchainConfig.rpcHost)}:${this.blockchainConfig.rpcPort}`);
```

Comment:
Should the `http` be hardcoded? I know this is existing code, but might be a good time to update if needed.